📝 Fix document

munechika-koyo opened the pull request:
## Issues
- Some API documentation contains typos or is incomplete.
#455 
- The latest `sphinx_rtd_theme` is incompatible with the current style settings.
#456
- Obsolete installation guide
#458 
- Some RST files are not referenced because they seem stale.
- Entries at excessively deep levels are displayed in the ToC tree.

## TODO list
- [x] Corrected typos in class references and added missing classes.
#455 
- [x] Introduced custom CSS for responsive table styling and image aspect ratio preservation.
- [x] Removed previously overridden theme CSS.
- [x] Re-configure Sphinx settings.
#456
- [x] Upgrade installation procedure
#458
- [x] Remove unused RST files.
- [x] Refine toctree directives.
- [x] Add a document about `LoggingRay`

## Preview demo site
The modified documentation site preview is available here:
https://munechika-koyo.github.io/source/

munechika-koyo committed:
Fix typos in class references and add some missing classes

munechika-koyo committed:
Add custom CSS for responsive table styling and image aspect ratio preservation

munechika-koyo committed:
Remove previous overriden theme css

munechika-koyo commented:
Displaying version at the top of the sidebar seems deprecated since v3: https://sphinx-rtd-theme.readthedocs.io/en/stable/configuring.html#confval-display_version

munechika-koyo committed:
Update Sphinx configuration to set default pygments style and specify HTML theme options

munechika-koyo committed:
Update Sphinx configuration to include custom static path and simplify CSS file reference

munechika-koyo commented:
BSDF-related codes were deleted (#349), but some documents still remain:
https://github.com/raysect/source/blob/2487a86b34aea3c7e095aee80e7eb2ebb0c3c983/docs/source/demonstrations/demonstrations.rst?plain=1#L87-L96

The missing demo scripts trigger warnings during a document build like this:
```bash
WARNING: Include file '.../source/demos/reflectivity/plotting_brdfs.py' not found or reading it failed [docutils]
WARNING: Include file '.../source/demos/reflectivity/plotting_brdfs_3d.py' not found or reading it failed [docutils]
```
How should we handle this?

munechika-koyo commented:
Additionally, there are many outdated RST files visible:
```bash
source/docs/source/api_reference/core/raysect_core_kdtree.rst: WARNING: document isn't included in any toctree [toc.not_included]
source/docs/source/architecture.rst: WARNING: document isn't included in any toctree [toc.not_included]
source/docs/source/demonstrations/core/energy_conservation.rst: WARNING: document isn't included in any toctree [toc.not_included]
source/docs/source/demonstrations/core/multiple_importance_sampling.rst: WARNING: document isn't included in any toctree [toc.not_included]
source/docs/source/demonstrations/core/ray_intersection_points.rst: WARNING: document isn't included in any toctree [toc.not_included]
source/docs/source/demonstrations/core/world_contains_point.rst: WARNING: document isn't included in any toctree [toc.not_included]
source/docs/source/demonstrations/materials/anisotropic_emitters.rst: WARNING: document isn't included in any toctree [toc.not_included]
source/docs/source/demonstrations/materials/diamond.rst: WARNING: document isn't included in any toctree [toc.not_included]
source/docs/source/demonstrations/materials/diffuse_colours.rst: WARNING: document isn't included in any toctree [toc.not_included]
source/docs/source/demonstrations/materials/emissive_colours.rst: WARNING: document isn't included in any toctree [toc.not_included]
source/docs/source/demonstrations/materials/glass_bunny.rst: WARNING: document isn't included in any toctree [toc.not_included]
source/docs/source/demonstrations/materials/metal_materials.rst: WARNING: document isn't included in any toctree [toc.not_included]
source/docs/source/demonstrations/materials/polar_brdf_plots.rst: WARNING: document isn't included in any toctree [toc.not_included]
source/docs/source/demonstrations/materials/surface3d_brdf_plots.rst: WARNING: document isn't included in any toctree [toc.not_included]
source/docs/source/demonstrations/materials/surface_roughness.rst: WARNING: document isn't included in any toctree [toc.not_included]
source/docs/source/demonstrations/materials/tetrahedra_mesh_emitter.rst: WARNING: document isn't included in any toctree [toc.not_included]
source/docs/source/demonstrations/materials/volume_inhomogeneous.rst: WARNING: document isn't included in any toctree [toc.not_included]
source/docs/source/demonstrations/observers/animations.rst: WARNING: document isn't included in any toctree [toc.not_included]
source/docs/source/demonstrations/observers/cornell_box.rst: WARNING: document isn't included in any toctree [toc.not_included]
source/docs/source/demonstrations/observers/cornell_box_with_camera.rst: WARNING: document isn't included in any toctree [toc.not_included]
source/docs/source/demonstrations/observers/mesh_observers.rst: WARNING: document isn't included in any toctree [toc.not_included]
source/docs/source/demonstrations/observers/metal_balls_with_lens.rst: WARNING: document isn't included in any toctree [toc.not_included]
source/docs/source/demonstrations/observers/optical_fibre.rst: WARNING: document isn't included in any toctree [toc.not_included]
source/docs/source/demonstrations/observers/orthographic_camera.rst: WARNING: document isn't included in any toctree [toc.not_included]
source/docs/source/demonstrations/optics/etendue_of_pinhole.rst: WARNING: document isn't included in any toctree [toc.not_included]
source/docs/source/demonstrations/optics/prism.rst: WARNING: document isn't included in any toctree [toc.not_included]
source/docs/source/quickstart_more.rst: WARNING: document isn't included in any toctree [toc.not_included]
source/docs/source/sampling.rst: WARNING: document isn't included in any toctree [toc.not_included]
```

munechika-koyo committed:
Update installation instructions to reflect Python 3.9+ requirement and remove scipy and python v2 mention

munechika-koyo committed:
Update installation instructions to use 'python -m pip' for installation commands and clarify source installation process

munechika-koyo committed:
Fix typo: change 'demo' to 'demos'

munechika-koyo committed:
Remove unnecessary references to GitHub repository and Python Setup Tools in installation documentation

munechika-koyo committed:
Update testing instructions to use Python's built-in unittest framework

munechika-koyo committed:
Fix formatting in installation documentation and enhance unittest framework reference

CnlPepper commented:
Thanks for working to update the docs. I didn't have time to go through them. Feel free to clean up any stale files, they will be in the repository history. Happy to merge this when you are ready.

munechika-koyo commented:
I got it! @CnlPepper 
I would try to delete all obsolete files.

Regarding [BRDF plotting demo issue](https://github.com/raysect/source/pull/457#issuecomment-3164478081), I would also try to figure out how to reproduce demo scripts.

munechika-koyo committed:
Remove unnecessary VSCode settings file

CnlPepper committed:
Merge pull request #460 from raysect/development

Release v0.9.1

munechika-koyo committed:
Merge remote-tracking branch 'upstream/master' into fix/document

munechika-koyo committed:
Merge branch 'development' into fix/document

munechika-koyo committed:
Remove obsolete documentation files

munechika-koyo committed:
Add :titlesonly: directive to API reference documentation

munechika-koyo committed:
Enhance LoggingRay docstrings

munechika-koyo committed:
Refactor imports in logging_trajectories.py for clarity and remove unused modules

munechika-koyo committed:
Add LoggingRay class to API documentation

munechika-koyo committed:
Add demonstration for Ray Logging Trajectories with visualizations

munechika-koyo committed:
Fix docstrings

Apply correct types, Fix error strings and Fix error indent

munechika-koyo committed:
Fix duplication of links and targets

munechika-koyo committed:
Fix not-included warning

demo docs are referenced with the internal hyperlink, so add `:orphan:` to remove not-included into toctree warning.

munechika-koyo commented:
There are still warnings about cross-referencing classes, as shown below.
Their classes are located in different modules under `raysect.core` or `raysect.optical`.
Removing these warnings requires a precise specification like `~raysect.core.ray.Ray`, not just `Ray`, for each type declaration in the docstring.
Because it is tedious and loses simplicity, we should keep them as warnings.
```bash
WARNING: more than one target found for cross-reference 'Ray': raysect.core.ray.Ray, raysect.optical.ray.Ray [ref.python]
WARNING: more than one target found for cross-reference 'World': raysect.core.scenegraph.world.World, raysect.optical.scenegraph.world.World [ref.python]
WARNING: more than one target found for cross-reference 'Function1D': raysect.core.math.function.float.function1d.base.Function1D, raysect.core.math.function.vector3d.function1d.base.Function1D [ref.python]
WARNING: more than one target found for cross-reference 'Function2D': raysect.core.math.function.float.function2d.base.Function2D, raysect.core.math.function.vector3d.function2d.base.Function2D [ref.python]
WARNING: more than one target found for cross-reference 'Function3D': raysect.core.math.function.float.function3d.base.Function3D, raysect.core.math.function.vector3d.function3d.base.Function3D [ref.python]
```
Note that it might be necessary if we need to guide users to correct API definitions.

munechika-koyo committed:
Remove trailing blank lines

munechika-koyo committed:
Add intersphinx extension and configuration for external documentation links

munechika-koyo committed:
Add :titlesonly: option for math docs

munechika-koyo committed:
Add plotting script for Aluminium BRDF and update documentation reference

munechika-koyo committed:
Add 3D surface BRDF plotting script and update documentation reference

munechika-koyo committed:
Refactor BRDF plotting demo to use subplots for multiple light angles

munechika-koyo committed:
Update BRDF surface 3D plots image created by renewed demo

munechika-koyo commented:
There are still some warnings, except for those mentioned above ([build log](https://github.com/munechika-koyo/source/actions/runs/17379271315/job/49332657649)). However, this PR should be ready for review @CnlPepper .

The other warning:
```bash
WARNING: duplicate object description of raysect.primitive.Sphere, other instance in api_reference/primitives/geometric_primitives, use :no-index: for one of them
...
```
seems to be related to the fact that the `Sphere` class API docs are generated not only for [API pages](https://munechika-koyo.github.io/source/api_reference/primitives/geometric_primitives.html) but also for the [primitive page](https://munechika-koyo.github.io/source/primitives.html).

munechika-koyo committed:
Fix indentation in LoggingRay usage examples for consistency
